MyArrayList.java

```diff
@@ -1 +1,100 @@
+package com.company;
 
+public class MyArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private int position = 0;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private StringBuilder stringBuilder = new StringBuilder();
+    private int size = array.length;
+
+    public void printArrayList() {
+        System.out.println("My array list: ");
+        stringBuilder.setLength(0);
+        for (int j = 0; j < size; j++) {
+            stringBuilder.append(array[j]).append(", ");
+        }
+        System.out.println(stringBuilder);
+    }
+
+    @Override
+    public void add(T value) {
+        check(position);
+        add(value, position);
+        position++;
+    }
+
+    private void check(int i) {
+        if (i >= size) {
+            size += DEFAULT_CAPACITY / 2;
+            array = newArray(array);
+        }
+    }
+
+    private Object[] newArray(Object[] newArray) {
+        newArray = new Object[size];
+        System.arraycopy(array, 0, newArray, 0, position);
+        return newArray;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        check(index);
```

Comment:
better add checking:
if(index<0 || index>position)...

Reply:
I added it in method check()